ui/app/AppLayouts/Onboarding2/KeycardCreateReplacementFlow.qml

```diff
@@ -0,0 +1,143 @@
+SQUtils.QObject {
+    id: root
+
+    required property StackView stackView
+
+    required property int keycardState
+    required property int addKeyPairState
+    required property int keycardPinInfoPageDelay
+
+    required property var isSeedPhraseValid
+
+    property bool displayKeycardPromoBanner
+
+    signal loginWithKeycardRequested
+    signal keycardFactoryResetRequested
+    signal keyPairTransferRequested
+    signal keycardPinCreated(string pin)
+    signal seedphraseSubmitted(string seedphrase)
+
+    signal keypairAddTryAgainRequested
+    signal reloadKeycardRequested
+    signal createProfileWithoutKeycardRequested
+
+    signal finished
+
+    function init() {
+        root.stackView.push(d.initialComponent())
+    }
+
+    QtObject {
+        id: d
+
+        function initialComponent() {
+            if (root.keycardState === Onboarding.KeycardState.Empty)
+                return seedphrasePage
+
+            if (root.keycardState === Onboarding.KeycardState.NotEmpty)
+                return keycardNotEmptyPage
+
+            return keycardIntroPage
+        }
+    }
+
+    Component {
+        id: keycardIntroPage
+
+        KeycardIntroPage {
+            keycardState: root.keycardState
+            displayPromoBanner: root.displayKeycardPromoBanner
+
+            onReloadKeycardRequested: {
+                root.reloadKeycardRequested()
+                root.stackView.replace(d.initialComponent(),
+                                       StackView.PopTransition)
+            }
+
+            onKeycardFactoryResetRequested: root.keycardFactoryResetRequested()
+            onEmptyKeycardDetected: root.stackView.replace(seedphrasePage)
+            onNotEmptyKeycardDetected: root.stackView.replace(keycardNotEmptyPage)
+        }
+    }
+
+    Component {
+        id: keycardNotEmptyPage
+
+        KeycardNotEmptyPage {
+            onReloadKeycardRequested: {
+                root.reloadKeycardRequested()
+                root.stackView.replace(d.initialComponent(),
+                                       StackView.PopTransition)
+            }
+
+            onLoginWithThisKeycardRequested: root.loginWithKeycardRequested()
+            onKeycardFactoryResetRequested: root.keycardFactoryResetRequested()
+        }
+    }
+
+    Component {
+        id: seedphrasePage
+
+        SeedphrasePage {
+            title: qsTr("Enter recovery phrase of lost Keycard")
+
+            isSeedPhraseValid: root.isSeedPhraseValid
+            onSeedphraseSubmitted: (seedphrase) => {
+                root.seedphraseSubmitted(seedphrase)
+                root.stackView.push(keycardCreatePinPage)
+            }
+        }
+    }
+
+    Component {
+        id: keycardCreatePinPage
+
+        KeycardCreatePinPage {
+            onKeycardPinCreated: (pin) => {
+                Backpressure.debounce(root, root.keycardPinInfoPageDelay, () => {
+                    root.keycardPinCreated(pin)
+                    root.keyPairTransferRequested()
```

Comment:
Just an FYI, those are made async in my PR. So if you ever work on other kaycard flows, just keep it in mind :smile: 

I'll take care of it when integrating.

Reply:
So the API needs to be revised. Isn't some additional state, similar to https://github.com/status-im/status-desktop/blob/0445f131f77cf5669c8b7200ae8be5cda3a09f36/ui/StatusQ/src/onboarding/enums.h#L48 needed then?